### DISCUSSION TITLE
Copilot sometimes get trapped in a long loop of repetitive suggestions

### DISCUSSION THREAD

**Discussion by RikEnde** [community/community#11377] [Copilot Conversations]
I'm not allowed / able to share the code that triggered these suggestions because it doesn't belong to me. Occasionally the copilot goes off on a tangent of what looks like repetitive iterations of the previous line or block:

![copilot-nonsense1](https://user-images.githubusercontent.com/6178351/153620733-b9081db6-1e28-4d7a-9a4c-2d543cef36fd.png)
<img width="456" alt="copilot-nonsense2" src="https://user-images.githubusercontent.com/6178351/153620740-89685084-51ca-421f-aa76-7e6faa2b6e72.png">

**Comment by KTibow** [comment]
I wonder how this could be reduced. You could try to make GitHub reject repetitive Codex suggestions, but it might be hard to see how "repetitive" a suggestion is.

This might also be reduced by increasing the temperature (essentially how random the output is, lower means the same thing over and over again, higher means random words).

**Reply by RikEnde** [reply]
Has anyone ever seen a "long" copilot suggestion that did make sense? Quite often I start typing a line of code, and copilot just finishes my thought, character for character exactly as it was in my head, but most of the time when the suggestion is more than half a dozen lines, it doesn't much sense.

**Reply by PrimeTheFirst** [reply]
@RikEnde I have. Copilot just writes the full function for me, and it makes sense. Copilot wrote this for a api in one go (though a element with id result didn't exist! ): 
![image](https://user-images.githubusercontent.com/89402723/154830628-eaf3db25-56d4-4c70-9add-c9b1e42a2e00.png)
But again, copilot gets stuck in loops quite a lot, after which I have to snap it out of it by writing more complex comments.

**Comment by ken-sands** [comment]
Just had similar with it producing the  following (and it went on and on until I guess it hit a text length limit)

 ```
 //validate the document is not expired
                        if($doc->expiry_date==null||$doc->expiry_date==''){
                            //no expiry date set
                            //check if document is not expired
                            if($doc->expiry_date==null||$doc->expiry_date==''){
                                //no expiry date set
                                //check if document is not expired
                                if($doc->expiry_date==null||$doc->expiry_date==''){
                                    //no expiry date set
                                    //check if document is not expired
                                    if($doc->expiry_date==null||$doc->expiry_date==''){
                                        //no expiry date set
                                        //check if document is not expired
                                        if($doc->expiry_date==null||$doc->expiry_date==''){
                                            //no expiry date set
                                            //check if document is not expired
                                            if($doc->expiry_date==null||$doc->expiry_date==''){
                                                //no expiry date set
                                                //check if document is not expired
                                                if($doc->expiry_date==null||$doc->expiry_date==''){
```
